docs/src/main/asciidoc/writing-native-applications-tips.adoc

```diff
@@ -183,9 +191,25 @@ public class MyClass {
 }
 ----
 
-==== Using a configuration file
+If you cannot modify the class (e.g. it is in a 3rd party jar) then you can instead specify the classes
+that you actually want to register in the `targets` member of the annotation:
+
+[source,java]
+----
+@RegisterForReflection(targets={ThirdPartyClass.class, OtherClass.class})
+public class EmptyClass {
+}
+----
+
+In this example only the `ThirdPartyClass` and `ThirdPartyClass` other class are registered, `EmptyClass` is ignored.
```

Comment:
I find that a bit counter intuitive. To be fair the whole annotation approach feels counter intuitive. I wonder if we could allow the class enlistment via an API and annotated toe block of code to be called.

```
@QuarkusBuild
public void buildOperations(QuarkusBuildOperations ops) {
    ops.enlistClassForReflecton(ThirdPartyClass.clas);
    ops.otherOperations(...);
}
```

Reply:
Grouping those operations into a single interface looks like a nice idea.

Reply:
This is actually pretty common in Jakarta EE (annotations that are not directly applicable to the class they are placed on). Some examples:

CDI response producer fields:

````
@Produces
@EntityManager
EntityManager entityManger;
````

This enables EntityManager injection in you application, but it can be literally anywhere and the class it is on makes no difference. Same with things like @DataSourceDefinition.

I do agree that it is a bit weird but your solution does not solve the problem of code affecting the whole app that could be anywhere.

Reply:
Also we would need both, as for the cases where the class is in your application then an annotation is the easiest to understand.